addons/delivery/models/stock_picking.py

```diff
@@ -187,12 +187,15 @@ def _send_confirmation_email(self):
     def _pre_put_in_pack_hook(self, move_line_ids):
         res = super(StockPicking, self)._pre_put_in_pack_hook(move_line_ids)
         if not res:
-            if self.carrier_id:
-                return self._set_delivery_package_type()
+            if move_line_ids.carrier_id:
+                if len(move_line_ids.carrier_id) > 1 or any(not ml.carrier_id for ml in move_line_ids):
+                    # avoid (duplicate) costs for products
+                    raise UserError(_("You cannot pack products into the same package when they have different carriers (i.e. check that all of their transfers have a carrier assigned and are using the same carrier)."))
```

Comment:
Are you sure this is *fun* enough? :upside_down_face:

Reply:
:shushing_face: :shushing_face: :shushing_face: :shushing_face:

Not having the feature in the first place was fun enough hashtag bitter that my original warnings were ignored and it was added in